monitoring/kernel.go

```diff
@@ -0,0 +1,103 @@
+/*
```

Comment:
If we move kernel-specific functionality into a separate file and it's still relying on Linux-specific code (eg `realKernelVersionReader`) - the source file needs to be Linux-specific and suffixed with `_linux.go` (same for test file) - otherwise we're risking build failure during publish step on darwin.